elfi/methods/diagnostics.py

```diff
@@ -0,0 +1,294 @@
+
+    def __init__(self, simulator, fn_distance, list_ss=None, prepared_ss=None,
+                 max_cardinality=4, seed=0):
+        """Initialise the summary-statistics selection for the Two Stage Procedure.
+
+        Parameters
+        ----------
+        simulator : elfi.Node
+            Node (often elfi.Simulator) for which the summary statistics will be applied.
+            The node is the final node of a coherent ElfiModel (i.e. it has no child nodes).
+        fn_distance : str or callable function
+            Distance metric (can be a string if the metric is implemented in ELFI).
+        list_ss : List of callable functions, optional
+            List of uncombined candidate summary statistics.
+        prepared_ss : List of lists of callable functions, optional
+            List of combined candidate summary statistics.
+        max_cardinality : int, optional
+            Maximum cardinality of a candidate summary-statistics combination.
+        seed : int, optional
+
+        """
+        if list_ss is None and prepared_ss is None:
+            raise ValueError('No summary statistics to assess.')
+
+        self.simulator = simulator
+        self.fn_distance = fn_distance
+        self.seed = seed
+        if prepared_ss is not None:
+            self.ss_candidates = prepared_ss
+        else:
+            self.ss_candidates = self._combine_ss(list_ss, max_cardinality=max_cardinality)
+        # Initialising an output pool as the rejection sampling will be used several times.
+        names_output = simulator.parents
+        names_output.append(simulator.name)
+        self.pool = elfi.OutputPool(names_output)
+
+    def _combine_ss(self, list_ss, max_cardinality):
+        """Create all combinations of the initialised summary statistics up till the maximum cardinality.
+
+        Parameters
+        ----------
+        list_ss : List of callable functions
+            List of uncombined candidate summary statistics.
+        max_cardinality : int
+            Maximum cardinality of a candidate summary-statistics combination.
+
+        Returns
+        -------
+        List
+            Combinations of candidate summary statistics.
+
+        """
+        if max_cardinality > len(list_ss):
+            max_cardinality = len(list_ss)
+
+        # Combine the candidate summary statistics.
+        combinations_ss = []
+        for i in range(max_cardinality):
+            for combination in combinations(list_ss, i + 1):
+                combinations_ss.append(combination)
+        return combinations_ss
+
+    def run(self, n_sim, n_acc, n_closest, batch_size=1, k=4):
+        """Run the Two Stage Procedure for identifying relevant summary statistics.
+
+        Parameters
+        ----------
+        n_sim : int
+            Number of the total ABC-rejection simulations.
+        n_acc : int
+            Number of the accepted ABC-rejection simulations.
+        n_closest : int
+            Number of the `closest' datasets
```

Comment:
Some reasonable default value would be user-friendly.

Reply:
Set a default value for `n_closest` similarly as for `n_sim`.